src/libraries/Microsoft.PowerFx.Core/IR/IRTranslator.cs

```diff
@@ -104,7 +105,26 @@ public override IntermediateNode Visit(TexlTableNode node, IRTranslatorContext c
                 Contracts.AssertValue(context);
 
                 var children = node.Children.Select(child => child.Accept(this, context)).ToArray();
-                return MaybeInjectCoercion(node, new TableNode(context.GetIRContext(node), children), context);
+                var irContext = context.GetIRContext(node);
+
+                if (!Preview.FeatureFlags.TableSyntaxDoesntWrapRecords || children.First() is not RecordNode)
+                {
+                    // Let's add "Value:" here                    
+                    var recordNodes = new List<RecordNode>();                    
+
+                    foreach (var childNode in children)
+                    {
+                        var values = new Dictionary<DName, IntermediateNode> { { AllowedValuesMetadata.ValueColumnName, childNode } };
+                        var rt = new RecordType().Add(TableValue.ValueName, childNode.IRContext.ResultType);
+
+                        recordNodes.Add(new RecordNode(new IRContext(childNode.IRContext.SourceContext, rt), values));
+                    }
+
+                    children = recordNodes.ToArray();                    
+                    irContext = new IRContext(node.GetCompleteSpan(), irContext.ResultType);
+                }
+
+                return MaybeInjectCoercion(node, new TableNode(irContext, children), context);
```

Comment:
>new TableNode(irContext, children [](http://example.com/codeflow?start=49&length=33)

can we remove TableNode altogether and just replace with a call to Table() 

Ie, see how &&/And keywords becomes a call to And() function.  (search for BuiltinFunctionsCore.And)

Reply:
Done

Reply:
Call is good, but we can still remove TableNode